apps/worker/services/cleanup/regular.py

```diff
@@ -1,15 +1,121 @@
 import logging
 import random
 
+import sentry_sdk
 from django.db.models.query import QuerySet
+from shared.django_apps.core.models import Pull, PullStates
+from shared.storage.exceptions import FileNotInStorageError
 
 from services.cleanup.cleanup import run_cleanup
+from services.cleanup.models import DELETE_FILES_BATCHSIZE
 from services.cleanup.uploads import cleanup_old_uploads
 from services.cleanup.utils import CleanupResult, CleanupSummary, cleanup_context
 
 log = logging.getLogger(__name__)
 
 
+@sentry_sdk.trace
+def cleanup_flares(
+    context, batch_size=DELETE_FILES_BATCHSIZE, limit=10000
+) -> CleanupSummary:
+    """
+    Flare is a field on a Pull object.
+    Flare is used to draw static graphs (see GraphHandler view in api) and can be large.
+    The majority of flare graphs are used in pr comments, so we keep the (maybe large) flare "available"
+    in Archive storage while the pull is OPEN.
+    If the pull is not OPEN, we dump the flare to save space.
+    If we need to generate a flare graph for a non-OPEN pull, we build_report_from_commit
+    and generate fresh flare from that report (see GraphHandler view in api).
+
+    This will only update the _flare_storage_path field for pulls whose flare files were
+    successfully deleted from storage.
+    """
+    # For any Pull that is not OPEN, clear the flare field(s), targeting older data
+    non_open_pulls = Pull.objects.exclude(state=PullStates.OPEN.value).order_by(
+        "updatestamp"
+    )
+
+    log.info("Starting flare cleanup")
+
+    # Clear in db
+    non_open_pulls_with_flare_in_db = non_open_pulls.filter(
+        _flare__isnull=False
+    ).exclude(_flare={})
+
+    # Process in batches
+    total_db_updated = 0
+    start = 0
+    while start < limit:
+        stop = start + batch_size if start + batch_size < limit else limit
+        batch = non_open_pulls_with_flare_in_db.values_list("id", flat=True)[start:stop]
+        if not batch:
+            break
+        n_updated = non_open_pulls_with_flare_in_db.filter(id__in=batch).update(
```

Comment:
splitting into two queries does make more sense here, as it constrains the `update`. Which I don’t think you necessarily need to do, as that query might be trivial/fast enough to not matter.